### PR TITLE
Icon: remove BaseComponent usage.

### PR DESCRIPTION
**Pull request by dzearing**
Removes BaseComponent from Icon.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8373)

### CODE REVIEW

File: common/changes/office-ui-fabric-react/more-bc-removal_2019-03-19-04-21.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "Icon: remove BaseComponent dependency.",
```
Comment (KevinTCoughlin) [inline common/changes/office-ui-fabric-react/more-bc-removal_2019-03-19-04-21.json]:
```suggestion
      "comment": "Icon & Image: remove BaseComponent dependency.",
```

**Comment by JasonGore** [comment]
This is technically a breaking change because `BaseComponent` has `public` members that could break anybody using them via `Icon`. Also if someone was extending `Icon` and using `BaseComponent` `protected` members they would also see a break...

**Comment by KevinTCoughlin** [comment]
> This is technically an API breaking change because `BaseComponent` has `public` members that could break anybody using them via `Icon`. Also if someone was extending `Icon` and using `BaseComponent` `protected` members they would also see a break...

@dzearing I've had this concern in the past and we've let a few go in because they were not publicly exported. For the publicly exported types, should we target these changes for `7.x`? 

Its a bit unfortunate we'd have to wait until next `major`, but I don't see a way around it aside from supporting two flavors side-by-side in `6,x`.

**Comment by dzearing** [comment]
I believe this is OK here; afaik there are no BaseComponent methods that were intentionally accessible for external usage. There may have been things that were public indirectly; for example, state. Which would be a bug to reference from any place other than the React library.

**Pull request merged**